Sources/LiveKit/Core/Room.swift

```diff
@@ -315,28 +324,35 @@ public class Room: NSObject, ObservableObject, Loggable {
 
         try Task.checkCancellation()
 
-        _state.mutate { $0.connectionState = .connecting }
+        _state.mutate {
+            $0.url = url
+            $0.token = token
+            $0.connectionState = .connecting
+        }
 
         do {
-            try await fullConnectSequence(url, token)
-
-            // Connect sequence successful
-            log("Connect sequence completed")
+            while true {
+                let region = try await resolveBestRegion()
```

Comment:
would we really want to await region selection on initial connection? 
The other SDKs don't block here and just continue immediately in order for region manager not to have any impact on connection speed